File: typescript/src/model/identity.ts
export class Identity {
    constructor(
        public uid: string,
        public name: string | null = null,
        public siteTag: string | null = null,
        public alternateIds: string[] = [],
        public alternateNames: string[] = []
    ) { }

    toMap(): Record<string, any> {
        return {
            "UID": this.uid,
            "name": this.name,
            "siteTag": this.siteTag,
            "alternateIDs": this.alternateIds,
            "alternateNames": this.alternateNames,
        };
    }

    static fromMap(data: Record<string, any>): Identity {
        const uid = data["UID"];
        if (uid == null) {
            throw new Error('UID is not defined in the map');
        }
        const name = data["name"] ?? null;
        const siteTag = data["siteTag"] ?? null;
        const alternateIds = data["alternateIDs"] ?? [];
        const alternateNames = data["alternateNames"] ?? [];
        return new Identity(uid, name, siteTag, alternateIds, alternateNames);
    }
}

File: dart/lib/src/model/identity.dart
import 'package:either_dart/either.dart';
import 'package:openral_core/src/cross/backend/parsing/parsing_error.dart';
import 'package:openral_core/src/model_parser/identity_parser.dart';

class Identity {
  String uid;
  String? name;
  String? siteTag;
  List<String> alternateIDs;
  List<String> alternateNames;

  Identity({
    required this.uid,
    this.name,
    this.siteTag,
    this.alternateIDs = const [],
    this.alternateNames = const [],
  });

  Map<String, dynamic> toMap() {
    return {
      "UID": uid,
      "name": name,
      "siteTag": siteTag,
      "alternateIDs": alternateIDs,
      "alternateNames": alternateNames,
    };
  }

  static Either<Identity, ParsingError> fromMap(map) {
    final identityParser = IdentityParser();

    return identityParser.fromMap(map);
  }
}
